Initial commit (from labs)

nemchik opened the pull request:
Confs pulled from https://github.com/linuxserver/fail2ban-confs

cstate post https://github.com/linuxserver/cstate/pull/113

nemchik committed:
Initial commit (from labs)

nemchik committed:
Remove cron (unused)

nemchik committed:
Update documentation

nemchik committed:
Chain clarification

quietsy commented on root/defaults/fail2ban/jail.conf:
```diff
@@ -0,0 +1,982 @@
+
+[domino-smtp]
+port    = smtp,ssmtp
+logpath = /home/domino01/data/IBM_TECHNICAL_SUPPORT/console.log
```
Should the paths be standardized?

nemchik replied:
https://github.com/fail2ban/fail2ban/pull/3315 I'm hoping to standardize this upstream. I can implement these changes in our image, but I wanted to keep this file as close to upstream as possible to minimize effort needed to update.

This would not be the only line that needs changing. Thoughts?

quietsy replied:
I vote for completely excluding the default configs and only include what we want and are willing to support

Roxedus replied:
We do have https://github.com/linuxserver/fail2ban-confs, which i really think should be used the same way we do in swag.

nemchik replied:
> We do have https://github.com/linuxserver/fail2ban-confs, which i really think should be used the same way we do in swag.

i didn't even know we had that. I'll get it set up.

nemchik replied:
Confs moved in https://github.com/linuxserver/docker-fail2ban/pull/1/commits/63086e04801fddab0f20d32176ef85c3459f9636
Continuing discussions about including defaults here and on discord.

nemchik replied:
Remaining confs that are not included in https://github.com/linuxserver/fail2ban-confs as of writing can be added at a later date. Marking as resolved for now.

quietsy commented on Dockerfile.armhf:
```diff
@@ -0,0 +1,28 @@
+FROM ghcr.io/linuxserver/baseimage-alpine:arm32v7-3.16
```
Should we skip armhf since we dropped it officially?

nemchik replied:
I guess removing it rather than releasing and then deprecating in the near future (if necessary) would be better. I don't know how necessary it would be to deprecate, since fail2ban is just a python package. I don't mind either way.

nemchik replied:
For now we'll keep it. As long as Alpine has armhf available, and python3 is available on armhf.

quietsy commented on root/defaults/fail2ban/jail.conf:
```diff
@@ -0,0 +1,982 @@
+# SSH servers
+#
+
+[sshd]
```
What's the difference between jails defined here and under jail.d?

nemchik replied:
ref: https://github.com/crazy-max/docker-fail2ban#global-jail-configuration
```text
jail.conf
jail.d/*.conf (in alphabetical order)
jail.local
jail.d/*.local (in alphabetical order)
```
With actions and filters it's
```text
action.d/*.conf (in alphabetical order)
action.d/*.local (in alphabetical order)
```
and
```text
filter.d/*.conf (in alphabetical order)
filter.d/*.local (in alphabetical order)
```
respectively.

So basically, the `*.conf` files in any location are the out of the box configs, with everything pretty much disabled. The `*.local` are user created files where they can enable things and change values to suit their needs (and you don't need to copy the original `*.conf`, just modify the lines you need). And to directly answer your question, `jail.conf` is read first, then `jail.d/*.conf` (in alphabetical order) which is where I'm putting any added confs that are not part of the upstream fail2ban `jail.conf` unless they later get PR'd upstream and merged, they could then be removed from this repo.

The directory structure information could be included in the README (either in the root of the repo, or in the configuration readme placed in `/config/fail2ban/README.md`. I had meant to do that, but forgot. I'll add it shortly.

nemchik replied:
Resolved in https://github.com/linuxserver/docker-fail2ban/pull/1/commits/3bfb129ae732cc35ab9f7ce44a0fe432a9af5a39

nemchik committed:
Adjust docs

nemchik committed:
Move fail2ban confs to https://github.com/linuxserver/fail2ban-confs

nemchik committed:
Fix mkdir

thespad commented on Dockerfile:
```diff
@@ -0,0 +1,55 @@
+FROM ghcr.io/linuxserver/baseimage-alpine:3.16
```
Use 3.15 for now until 3.16 is fully switched over to s6 v3 otherwise it'll make your life difficult transitioning the container.

nemchik replied:
Good call

nemchik replied:
Resolved in https://github.com/linuxserver/docker-fail2ban/pull/1/commits/6878c98e634358f351f8f1f41e54e96ca859fdb0

thespad commented on Dockerfile:
```diff
@@ -0,0 +1,55 @@
+  curl -o \
+    /tmp/fail2ban-confs.tar.gz -L \
+    "https://github.com/linuxserver/fail2ban-confs/tarball/master" && \
+  tar xf \
```
Can you not just untar directly into the defaults directory? It looks like you're just copying 1:1 anyway.

thespad commented on Dockerfile:
```diff
@@ -0,0 +1,55 @@
+  curl -o \
+    /tmp/fail2ban-confs.tar.gz -L \
+    "https://github.com/linuxserver/fail2ban-confs/tarball/master" && \
+  tar xf \
+    /tmp/fail2ban-confs.tar.gz -C \
+    /tmp/fail2ban-confs --strip-components=1 && \
```
Can you not just untar directly into /defaults? It looks like you're copying 1:1 afterwards anyway.

nemchik replied:
I don't want the `.github` folder, but I guess I could remove that after extracting. I went with being (maybe too) specific.

thespad replied:
I don't suppose it has any real impact on build speed because it's still got to copy the same files, it's just less obvious what's happening at a glance.

nemchik replied:
Resolved in https://github.com/linuxserver/docker-fail2ban/pull/1/commits/6878c98e634358f351f8f1f41e54e96ca859fdb0

nemchik committed:
Use alpine 3.15
Extract confs directly to /defaults
Use alpine mirror from official list #203
Update Configuration README url

nemchik committed:
Fix tar

quietsy commented on README.md:
```diff
@@ -1 +1,305 @@
+services:
+  fail2ban:
+    image: lscr.io/linuxserver/fail2ban:latest
+    container_name: fail2ban
```
Should the example use host networking by default?
I can't imagine this being too useful without it

nemchik replied:
Good catch, and yes, you are correct. I thought I had it with https://github.com/linuxserver/docker-fail2ban/blob/initial/readme-vars.yml#L21

I'll go dig through Jenkins to find what actually does it.

nemchik replied:
Fixed in https://github.com/linuxserver/docker-fail2ban/pull/1/commits/b2e566deae964c6ee10a0f3fe0e57e808b4610d4

nemchik committed:
Fix readme

nemchik committed:
Update release date
